test/visual/mpl/circuit/test_circuit_matplotlib_drawer.py

```diff
@@ -64,19 +68,87 @@ def cwd(path):
         os.chdir(oldpwd)
 
 
-class TestMatplotlibDrawer(QiskitTestCase):
+class TestCircuitMatplotlibDrawer(QiskitTestCase):
     """Circuit MPL visualization"""
 
     def setUp(self):
         super().setUp()
-        self.circuit_drawer = TestMatplotlibDrawer.save_data_wrap(
+        self.circuit_drawer = TestCircuitMatplotlibDrawer.save_data_wrap(
             _matplotlib_circuit_drawer, str(self)
         )
 
+        if not os.path.exists(FAILURE_DIFF_DIR):
+            os.makedirs(FAILURE_DIFF_DIR)
+
+        if not os.path.exists(RESULTDIR):
+            os.makedirs(RESULTDIR)
+
     def tearDown(self):
         super().tearDown()
         mpl_close("all")
 
+    @staticmethod
+    def _image_path(image_name):
+        return os.path.join(RESULTDIR, image_name)
+
+    @staticmethod
+    def _reference_path(image_name):
+        return os.path.join(TEST_REFERENCE_PATH, image_name)
```

Comment:
Minor: I'd roughly prefer these just to be inlined into the call sites with the constants becoming `pathlib.Path` instances.  It'd be the same or less typing, even at the point of call:
```
self._reference_path("my_image.png")  # current
TEST_REFERENCE_PATH / "my_image.png"  # alternate
```

Reply:
Chose not to inline these for the moment - but I can see the reasoning.